site/content/tutorial/14-composition/04-optional-slots/text.md

```diff
@@ -0,0 +1,25 @@
+---
+title: Optional slots
+---
+
+In the previous example, the contact card rendered fallback text if a named slot was left empty. But for some slots, perhaps you don't want to render anything at all. We can do this by checking the properties of the special `$$slots` variable.
```

Comment:
> But for some slots, perhaps you don't want to render anything at all.

this line feels weird, because by default if nothing is passed, `<slot name="xxx" />` will render nothing.

probably can reword to explain, you want to render the wrapper `div` conditionally, depending on whether the slot is passed in, or maybe conditionally add class name to element with `class:className={$$slots.slotName}` if there's no slot passed in

Reply:
Thanks for the feedback -- I can see that this doesn't describe the benefit of `$$slots` very well. Your note on `class:className={$$slots.slotName}` also made me realize that I wasn't really explaining the full power of `$$slots`, so I revised the tutorial chapter into a fuller example that shows off both the conditional wrapper `div` and conditional `class:` directive.